### PR TITLE
Fixed bug where the product onSale was set to true when sale price was 0

### LINKED ISSUE #2524: [Product Details] [Price] Display sales price only when >$0
In the "price" row of the Product Details screen, ["Sales price" should only be displayed if there's a sales price set. ](https://d.pr/i/uryIWh)If not, let's only display "regular price".

### PR DESCRIPTION
**Pull request by anitaa1990**
This tiny PR fixes #2524 by adding logic to set `isOnSale` to true only if the sale price is > 0.

#### To Reproduce and Test
- Click on a product from the products tab with sale price = 0.
- Notice that sale price is not displayed.
- Click on the Pricing section.
- Modify the **regular price** field and click on `Done` menu button.
- Notice that the sale price is displayed even when it was not updated.
- Pull changes from this PR and follow the above steps. Notice that the sale price is NOT displayed.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt [hunk 1]
```diff
@@ -518,7 +519,8 @@ class ProductDetailViewModel @AssistedInject constructor(
         productPricingViewState = if (inputValue > regularPrice) {
             productPricingViewState.copy(salePriceErrorMessage = string.product_pricing_update_sale_price_error)
         } else {
-            updateProductDraft(salePrice = inputValue, isOnSale = true)
+            val isOnSale = !(inputValue isEqualTo BigDecimal.ZERO)
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt:522]: Tip: You can use `isNotEqualTo` function ;)

**Comment by anitaa1990** [comment]
Thanks for the review @Onko!

> When I set the price on the web I can still see the 0 sale price. Is that ok?

I am not able to reproduce this scenario. I tried changing the sale price to 0 in the web and refreshed it in the app and it is not visible. Is there a particular set of steps I need to follow to replicate this?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt]: Good point 👍 Fixed in isNotEqualTo

**Comment by anitaa1990** [comment]
Hey @AmandaRiu @ThomazFB, Ondrej was reviewing this PR but since he's unavailable this sprint, would one of you mind taking a look at this PR please?

**Pull request merged**